### PR TITLE
[WIP] fix: inline alignment formatting

### PR DESCRIPTION
**Pull request by mgrip**
https://github.com/prettier/plugin-php/issues/394

### CODE REVIEW

File: src/clean.js [hunk 1]
```diff
@@ -119,6 +119,11 @@ function clean(node, newObj) {
     });
   }
 
+  // for inline code, we always use the long form <?php
+  if (node.kind === "echo") {
+    newObj.shortForm = false;
+  }
```
Comment (evilebottnawi) [inline src/clean.js]: It is not right, prettier should not change ast, many people use difference `echo` form in difference situation. We need output echo as is.

File: tests/inline/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -360,6 +360,84 @@ exports[`mixed-4.php 1`] = `
...
+    <?php echo $this->element(
+        'collection/item-title',
+        [
+            'class' => $itemTitleClass,
+            'small_title' => $item->small_title,
+            'title' => $item->title,
+            'url' => $url,
+            'color' => $item->title_color ?: 'primary'
+        ]
+    ); ?>
```
Comment (evilebottnawi) [inline tests/inline/__snapshots__/jsfmt.spec.js.snap]: We should use there `group`.

I.E. if it is allow be in one line print one line:
```php
<?php echo 1; ?>
```

If something break - break all (add newline after `<?php` and before `?>`) (as we do in prettier :smile: ):
```php
<?php
echo $this->veryVeryVeryVeryVeryVeryVeryLongFunction(
    'something',
   [
      'class' => $itemTitleClass
      // etc
   ]
);
?>
```

File: src/clean.js (hunk 1 above)
Comment (mgrip) [reply src/clean.js]: fine by me 👍

**Comment by mgrip** [comment]
thanks for the feedback @evilebottnawi! i can try and take another look again next week, but feel free to make updates on the branch if you'd like in the meantime!

**Comment by mgrip** [comment]
@evilebottnawi any preference on a gameplan for this? should we continue this PR or are you still planning on doing a separate one?

**Comment by evilebottnawi** [comment]
@mgrip don't have time right now, sorry, regular work, can we separate this pr on three parts as i wrote above?

**Comment by mgrip** [comment]
no worries! just checking. and what do you mean exactly by separating? I was at least trying to only address `Support indent in <?php and ?> (issue #394)`

**Comment by evilebottnawi** [comment]
@mgrip hm, let's fix all problems which i describe above :+1:

File: tests/inline/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (mgrip) [reply tests/inline/__snapshots__/jsfmt.spec.js.snap]: @evilebottnawi I've been digging into this and I'm unsure if its going to be possible...

Inline nodes and regular php nodes are stored next to each other in a flat list, there's no top level node to group. ie if you had something like this

```php
<div>Foo</div>
<?php
$a = 1;
$some_otherThing = 1;
?>
<span>test</span>
```

the node list ends up being

```
[
  {type: "inline" ...},
  {type: "assign" ...},
  {type: "assign" ...},
  {type: "inline" ...}
]
```

So if we wanted to group the php open/close tags with the assign nodes, there's no parent node to actually group them together. Any ideas?

Comment (mgrip) [reply tests/inline/__snapshots__/jsfmt.spec.js.snap]: We could maybe do something hacky our parser file to look through the ast we get from `php-parser` and create a faux top-level node for anything surrounded by inline nodes? Thats the only think I can think of atm

Comment (evilebottnawi) [reply tests/inline/__snapshots__/jsfmt.spec.js.snap]: @mgrip Let's solve this in `3. Improve group for <?php and ?>`. Right now let's left this as is

Comment (mgrip) [reply tests/inline/__snapshots__/jsfmt.spec.js.snap]: sounds good 👍

File: src/printer.js [hunk 3]
```diff
@@ -76,7 +77,29 @@ function genericPrint(path, options, print) {
       comments.printDanglingComments(path, options, /* sameIndent */ true)
     ]);
   }
-  const printed = printNode(path, options, print);
+  // if this is embedded code, lets try to match the indentation of the open <?php tag
+  let printed;
+  if (isPrevNodeInline(path)) {
+    const tagOpenIndex = options.originalText.lastIndexOf(
+      "<",
```
Comment (evilebottnawi) [inline src/printer.js]: Let's add test: `<?php echo '<'; ?><?php echo '<'; ?>`. I don't sure what using only `<` is good solution

**Comment by mgrip** [comment]
awesome 👍 ill try to get the ast change and `"<"` issues fixed tomorrow!

**Comment by mgrip** [comment]
After more investigation i think something like the solution i outlined above might be necessary to work - i was only looking at examples with 1 node, but for any inline code with multiple statements, we really need a parent grouping for the `align()` to work...

What do you guys think about something like this?

> We could maybe do something hacky in our parser file to look through the ast we get from php-parser and create a faux top-level node for anything surrounded by inline nodes? Thats the only think I can think of atm

**Comment by mgrip** [comment]
Opened this on the parser side just to start a discussion https://github.com/glayzzle/php-parser/issues/159

**Comment by mgrip** [comment]
Unfortunately this turned out to be more difficult than I thought - I'm still unsure of the best way to solve this...

**Comment by mgrip** [comment]
eclipsed by https://github.com/prettier/plugin-php/pull/431

**Pull request closed without merging**